### PR TITLE
Add ability to use mailboxes outside of grid actors.

### PR DESCRIPTION
**Pull request by mdmarek**
This PR is based on changed from actually trying to use grid in our internal products. The basic changes are: *A mailbox needs a server to run, so have that be its parameter rather than a context*

The rest of the changes are document updates, better logging, and a couple more tests.

### CODE REVIEW

File: grid.v3/cfg_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package grid
+
+import "testing"
+import "time"
+
+func TestSetClientCfgDefaults(t *testing.T) {
```
Comment (epsniff) [inline grid.v3/cfg_test.go]: :) thanks for doing this.

File: grid.v3/client.go [hunk 2]
```diff
@@ -300,16 +299,18 @@ func (c *Client) getWireClient(ctx context.Context, nsReceiver string) (WireClie
 	return cc.client, nil
 }
```
Comment (epsniff) [inline grid.v3/client.go]: :) I originally called it Minus from the SQL operator from the same name.   difference also fine.

File: grid.v3/mailbox.go [hunk 3]
```diff
@@ -25,31 +22,55 @@ func (box *Mailbox) String() string {
...
+//
+// Using a mailbox requires that the process creating the mailbox also
+// started a grid Server.
+func NewMailbox(s *Server, name string, size int) (*Mailbox, error) {
+	if !isServerRunning(s) {
+		return nil, ErrServerNotRunning
+	}
 	if !isNameValid(name) {
 		return nil, ErrInvalidMailboxName
 	}
```
Comment (epsniff) [inline grid.v3/mailbox.go]: reminds me of matchbox: https://dzone.com/articles/lose-the-lock-while-embracing-concurrency-brave-ne

File: grid.v3/client.go (hunk 2 above)
Comment (mdmarek) [inline grid.v3/client.go]: @epsniff Soooooorry, yes that's been there forever in SQL, totally didn't think about that... changed it back to minus!

**Pull request merged**